### ISSUE TITLE
which attention architecture is used in NER?

### ISSUE THREAD

**Issue by omerarshad** [IsaacChanghau/neural_sequence_labeling#6]
I want to understand how you used attention in NEr task, any paper or article which explains this? Thanks

**Comment by qq547276542** [comment]
According to the README, attention mechanism not suitable for NER task：
The variant modules include Stack Bidirectional RNN (multi-layers), Multi-RNN Cells (multi-layers), Lurong/Bahdanau Attention Mechanism, Self-attention Mechanism, Residual Connection, Layer Normalization and so on. However, these modifications did not improve (sometime even worse than basement model) the performance significantly (F1 score improves >= 1.5). It's easy to apply those variants and train by modifying the config settings in train_conll_ner_blstm_cnn_crf.py.

**Comment by qq547276542** [comment]
In my own experiment, attention mechanism really not work, but Layer Normalization can improve the robustness of the model

**Comment by omerarshad** [comment]
well in my experiments attention only models achieve comparable results to LSTM, even got better than LSTM with very less training time

**Comment by qq547276542** [comment]
> well in my experiments attention only models achieve comparable results to LSTM, even got better than LSTM with very less training time

Maybe there are some problems with my experiment.  I have tried BLSTM+SelfAttention+CRF，the effect is not as good as BLSTM+CRF.    The structure of your model is SelfAttention+CRF, not LSTM? I want to give it a try.

**Comment by omerarshad** [comment]
yes, structure of my model is attention+crf only

**Comment by qq547276542** [comment]
> yes, structure of my model is attention+crf only

do you have the relevant code, can I refer to it?

**Comment by MingLunHan** [comment]
I do not use crf, and get a best result.

**Comment by VioletJKI** [comment]
> I do not use crf, and get a best result.

@MingLunHan excuse me, what is your model's architecture? bilstm + attention only?

**Comment by MingLunHan** [comment]
> > I do not use crf, and get a best result.
> 
> @MingLunHan excuse me, what is your model's architecture? bilstm + attention only?

yes, i just used char representation, pretrained word embedding, bilstm and attention, CE loss. and its best result is 68.7, close to the performance in this project. However, crf got a worse result, about 67.XX...
